### PR TITLE
feat(dw): move project selector to page head

### PR DESCRIPTION
**Pull request by gitdallas**
Closes: [RHOAIENG-4623](https://issues.redhat.com/browse/RHOAIENG-4623)

## Description
Moved the project selector to the header for workload metrics (was previously at the top of each tab page section)

Before:
image

After:
image

## How Has This Been Tested?
Tested changes by checking:

- Project selector exists in header with both tabs selected
- Project selector does not exist in either tab anymore
- Selecting a project updates the data, reflecting the project has changed
- Selecting a project does not change the page/tab we are on

(operates in the same manner as previously, just shows up in a different area)

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx
Comment (mturley) [inline frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:10]: We shouldn't use anything from other `~/pages/*` subdirectories within here (a rule I know is broken elsewhere, but @andrewballantyne requested we stick to that for new code). Can we maybe move `ModelServingProjectSelection` into `~/concepts/projects/` and rename it something like `PageHeaderProjectSelection`, or just make our own `DistributedWorkloadsProjectSelection`?

Comment (andrewballantyne) [reply frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:10]: Ah man, this is getting fixed in the DS Projects rework PR... 🤦 We have two types of this component goofy like...

Maybe just update yours to match what he will do for the joys of the magic here? #2572 

https://github.com/opendatahub-io/odh-dashboard/blob/ad564f126f40e4372f76295c2a777134c43f184a/frontend/src/concepts/projects/ProjectSelector.tsx

Comment (mturley) [reply frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:10]: Ah, glad I tagged you then. Maybe we just hold off on this until after #2572 is merged

Comment (andrewballantyne) [reply frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:10]: You're welcome to take those exact changes over to your PR, I doubt they will change. Don't feel blocked by this effort.

File: frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx
Comment (mturley) [inline frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:37]: Come to think of it @gitdallas we could also remove all instances of the `projectSelectorMode` for these tabs since it'll always follow the `singleProjectOnly` logic now

File: frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx
Comment (mturley) [inline frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:38]: This comment can probably be removed because "either this tab requires one or we have a preferred one to go to" doesn't apply anymore and "No namespace is in the path" is obvious

File: frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx
Comment (mturley) [inline frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:43]: Same here, we can remove this or replace it with something like `// The namespace in the URL is invalid`

File: frontend/src/pages/distributedWorkloads/GlobalDistributedWorkloadsRoutes.tsx
Comment (mturley) [inline frontend/src/pages/distributedWorkloads/GlobalDistributedWorkloadsRoutes.tsx:23]: We do still need the `?` on `${tab.path}/:namespace?`, sorry, my mistake. It tells react-router that `:namespace` is optional here when matching the route (so that it will still render `GlobalDistributedWorkloads` when there's no namespace in the URL, which will catch that and redirect us). I had it confused with the `?` delimiter on the end of a URL for query params.

**Review by mturley — COMMENTED** [review]
A few small things

**Review by mturley — APPROVED** [review]
LGTM, although we probably want to just squash it to one commit

**Review by mturley — APPROVED** [review]

File: frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx
Comment (andrewballantyne) [inline frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:40]: What happens if you have no projects? Can you mount this component?

Comment (mturley) [reply frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:40]: See above `if (projects.length === 0) {`, it renders a "no projects" empty state

Comment (andrewballantyne) [reply frontend/src/pages/distributedWorkloads/global/GlobalDistributedWorkloads.tsx:40]: 🤦  yup -- for sure it is...

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**
